libbeat/common/transport/tlscommon/ca_pinning_test.go

```diff
@@ -44,22 +44,22 @@ var ser int64 = 1
 func TestCAPinning(t *testing.T) {
 	host := "127.0.0.1"
 
-	t.Run("when the ca_sha256 field is not defined we use normal certificate validation",
-		func(t *testing.T) {
-			cfg := common.MustNewConfigFrom(map[string]interface{}{
-				"certificate_authorities": []string{"ca_test.pem"},
-			})
+	t.Run("when the ca_sha256 field is not defined we use normal certificate validation", func(t *testing.T) {
+		cfg := common.MustNewConfigFrom(map[string]interface{}{
+			"verification_mode":       "strict",
+			"certificate_authorities": []string{"ca_test.pem"},
+		})
 
-			config := &Config{}
-			err := cfg.Unpack(config)
-			require.NoError(t, err)
+		config := &Config{}
+		err := cfg.Unpack(config)
+		require.NoError(t, err)
 
-			tlsCfg, err := LoadTLSConfig(config)
-			require.NoError(t, err)
+		tlsCfg, err := LoadTLSConfig(config)
+		require.NoError(t, err)
 
-			tls := tlsCfg.BuildModuleConfig(host)
-			require.Nil(t, tls.VerifyPeerCertificate)
-		})
+		tls := tlsCfg.BuildModuleClientConfig(host)
+		require.Nil(t, tls.VerifyConnection)
```

Comment:
About these tests (follow up): maybe it should be more tabular + do create a real certificate and check validation actually succeeds. VerifyConnection is a function and without actually running it we can't really tell if the function behaves as is presumed.

Reply:
The Golang test itself generates certificates to check if the pinning works. The python test uses a real certificate to see if the pinning works with the ES output. What else do you miss from these tests?

Reply:
the test only creates a configuration object, which can not be validated correctly. Better to have some real certificate validation here in order to verify that the configured *tls.Config really works as expected.

There should be no need for a python test.

Reply:
Created a follow-up issue: https://github.com/elastic/beats/issues/23581